ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -280,6 +293,113 @@ public synchronized Container getContainerById(int accountId, int containerId) t
     }
   }
 
+  /**
+   * Add a version of {@link Dataset} based on the supplied properties.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param accountName the name for the parent account.
+   * @param containerName the name for the container.
+   * @param datasetName the name of the dataset.
+   * @param version the version of the dataset.
+   * @param expirationTimeMs the expiration time of the version of the dataset.
+   * @return the {@link Dataset} which contains the metadata.
+   * @throws SQLException
+   */
+  public synchronized Dataset addDatasetVersions(int accountId, int containerId, String accountName,
+      String containerName, String datasetName, long version, long expirationTimeMs) throws SQLException {
+    Dataset dataset;
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      dataAccessor.getDatabaseConnection(true);
+      PreparedStatement insertDatasetVersionStatement =
+          dataAccessor.getPreparedStatement(insertDatasetVersionSql, true);
+      dataset = executeAddDatasetVersionStatement(insertDatasetVersionStatement, accountId, containerId, accountName,
+          containerName, datasetName, version, expirationTimeMs);
+      dataAccessor.onSuccess(Write, System.currentTimeMillis() - startTimeMs);
```

Comment:
can u put this metric in a finally clause ? we would emit it even if an error was thrown and it'd be interesting to see what latencies look like when there are errors. And is this latency primarily for measuring the time to write to db ? In that case, please move it closer to L316 just before the execute cmd.

Reply:
I could add some metrics in dataAccessor.onException just like what we did in dataAccessor.onSuccess, since if you put in finally clause, no matter it fail or Success we will measuring the time. And given it's a common change for all mysql queries, I would prefer to implement in another follow up pr.